docs/articles/licensing/gplv3/gplv3-blacklist.rst

```diff
@@ -0,0 +1,86 @@
+
+Modern cars have around 100 million lines of code running on them,12 with 70% 
+of that code being in the head unit. Complexity is a non-trivial issue in 
+automotive software design. In addition to being complex, cars can be dangerous.
+
+Locking the software to the hardware by signing the original software image
+with a cryptographic key so that only an image provided by the supplier will 
+boot or install – is a common practice in embedded devices.
+This process of signing software images – so only images with the right key 
+will boot or install effectively prevents a user from modifying the software
+on the device since they have no access to the key needed to allow their 
+modified version to boot or install.
+
+With regard to the GPLv3, the difficulty is with only the anti-tivoization 
+clause, and the reason for disfavoring that license is safety. 
+
+References
+^^^^^^^^^^
+
+`<https://www.gnu.org/licenses/gpl-faq.en.html#Tivoization>`_
+`<http://www.ifosslr.org/ifosslr/article/view/102/207>`_
+
+Difference between GPLv3 license usage of production and development images
+---------------------------------------------------------------------------
+
+* The development image can contain GPLv3 softwares even without commercial
+  license. E.g - gdbserver(GPLv3) can be used in development image 
+  but not in production image.
+* In production image if GPLv3 non-commercial licensed software is used,
+  then the vendor has to provide the necessary tools to change the 
+  software as well.
+
+
+How to facilitate blacklisting of GPLv3 licensed software in Yocto
+------------------------------------------------------------------
+
+1. Add layer **meta-gplv2**
+2. Remove "initramfs-live-install" and "initramfs-module-install" from 
+   environment variable PACKAGE_INSTALL.
+   *(file: core-image-minimal-initramfs.bbappend)*
+
+   .. code-block:: bash
+
+      PACKAGE_INSTALL_remove = "initramfs-live-install initramfs-module-install"
+3. Remove "grub-efi" from WKS_FILE_DEPENDS_BOOTLOADERS for x86-64 arch.
+   *(file: image_types_wic.bbclass)*
+
+   .. code-block:: bash
+
+     WKS_FILE_DEPENDS_BOOTLOADERS_remove_x86-64 = "grub-efi"
+4. Remove "parted" from RDEPENDS_${PN}. 
+   *(file: initramfs-module-install-efi_1.0.bb)*
+
+   .. code-block:: bash
+
+     RDEPENDS_${PN}_remove = "parted"
+5. Select "system-boot" as the bootloader instead of default "grub-efi".
+   *(file: systemd-boot.bbclass)*
+
+   .. code-block:: bash
+
+     EFI_PROVIDER = "systemd-boot"
+6. Deprecate GPLv3 and other unwanted licenses as follows - 
+
+   .. code-block:: bash
+
+	INCOMPATIBLE_LICENSE = "\
+	    GPL-3.0 \
+	    LGPL-3.0 \
+	    AGPL-3.0 \
+	    MPL-1.1 \
+	    SGI-1 \
+	    CECILL-2.0 \
+	    CDDL-1.1 \
+	    EPL-1.0 \
+	    CPL-1.0 \
+	"
+	PACKAGECONFIG_remove = "cairo"
+
+Impact of removing GPLv3 licensed software
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The software update feature won't work in PELUX, as it uses "parted" tool.
```

Comment:
We'd need a non-GPLv3 component unless partprobe is being used as a tool and not residing on the target. Is that the case?

Reply:
Unfortunately, it is actually part of the target so we need to get rid of it.
I tried a bunch of alternatives:
hdparm -z /dev/mmcblk0
sfdisk -R /dev/mmcblk0
blockdev --rereadpt /dev/mmcblk0
... But none of them would work

Reply:
If it is necessary for the core functionality of the software, I don't think we should remove it based on license. There are a number of alternatives for dealing with GPLv3, such as additional exceptions and modification under homologation (https://en.wikipedia.org/wiki/Homologation). 

In addition, the point of PELUX is to be an open source starting point for IVI, so we have no issues with GPLv3, in fact we like it since it is a better license than other strong copyleft licenses. While I think it is courteous of us to describe how you can manage the situation where a customer requests GPLv3 removal, it should not block the work we're doing in PELUX.

Reply:
Maybe we could keep partprobe then. For now, it is only needed to inform the kernel of disk changes after re-partitioning the raspberry pi's sdcard. I assume that "in production" we would have pre-partitioned disks that wouldn't require the usage of partprobe, so it's no big deal.